lib/puppet.rb

```diff
@@ -36,6 +50,29 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    if GETTEXT_AVAILABLE
+      # TODO: Handle the rubygems case?
+      local_locale_path = File.absolute_path('../locales', File.dirname(__FILE__))
+      posix_system_locale_path = File.absolute_path('../../../share/puppet/locale', File.dirname(__FILE__))
+      win32_system_locale_path = File.absolute_path('../../../../../puppet/share/puppet/locale', File.dirname(__FILE__))
+
+      if File.exist?(local_locale_path)
+        locale_path = local_locale_path
+      elsif File.exist?(win32_system_locale_path)
+        locale_path = win32_system_locale_path
+      elsif File.exist?(posix_system_locale_path)
+        locale_path = posix_system_locale_path
+      else
+        # We couldn't load our locale data. Possibly we're loaded as a rubygem or something?
+        locale_path = nil
+      end
+
+      if locale_path
+        GettextSetup.initialize(locale_path)
+        FastGettext.locale = GettextSetup.negotiate_locale(ENV["LANG"])
```

Comment:
So my thinking was, that the actual code we need is only 8 lines or so (A single API call) for Windows.  So we could just write that code internally instead of adding yet another gem dependency for something trivial.  That's why I though it would be better to refactor that into a method call (perhaps Puppet::Util is a bad idea, but could still be a separate method) so we can then test it appropriately.